### PR TITLE
Promised connect event

### PR DESCRIPTION
**Pull request by hadasiii**
According to documentation (https://node-postgres.com/api/pool) the pool 'onConnect' event is the place to run setup commands on a client. Unfortunately the _create function who's emitting the connect event does not wait on the connect to finish its work, which makes it impossible to run set up functionality like setting query group on the client or any async functionality for that matter.
The new code implements 'promise-events' as the EventEmitter. It inherits Node's built-in EventEmitter interface, except that selected methods are overridden to return a promise.
(https://www.npmjs.com/package/promise-events)
I've also added tests to check that we wait on the connect, plus catching an error if there is one.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -97,11 +97,16 @@ Pool.prototype._create = function (cb) {
...
+      return
     }
+
+    this.log('client connected')
+    this.emit('connect', client)
+      .then(() => {
+        this.log('came back from connect')
+        cb(null, client)
+      })
+      .catch(err => cb(err, null))
```
Comment (aviaoh) [inline index.js]: please add `this.log('connect listener error:', err)` before calling `cb`

File: test/events.js [hunk 2]
```diff
@@ -22,6 +22,44 @@ describe('events', function () {
...
   })
 
+  it('waits on connect until finished', function (done) {
+    var pool = new Pool()
+    var emittedClient = false
+    pool.on('connect', function (client) {
+      return client.query('SELECT now()')
+      .then(() => {
+        emittedClient = client
+      })
```
Comment (aviaoh) [inline test/events.js]: indentation 🙃 
(same for all `.then` in this file)

File: test/events.js [hunk 3]
```diff
@@ -22,6 +22,44 @@ describe('events', function () {
...
+    pool.connect(function (err, client, release) {
+      if (err) return done(err)
+      release()
+      pool.end()
+      expect(client).to.be(emittedClient)
+      done()
+    })
+  })
+
+  it('catch error on connect', function (done) {
```
Comment (aviaoh) [inline test/events.js]: please rename to catch error on connect listener.
the connection itself was done successfully in this test, the error was raised from the event listener

File: test/events.js [hunk 4]
```diff
@@ -22,6 +22,44 @@ describe('events', function () {
     })
   })
 
+  it('waits on connect until finished', function (done) {
+    var pool = new Pool()
+    var emittedClient = false
+    pool.on('connect', function (client) {
+      return client.query('SELECT now()')
```
Comment (aviaoh) [inline test/events.js]: why do you need to actually call query? looks irrelevant for this test.

File: test/events.js [hunk 5]
```diff
@@ -22,6 +22,44 @@ describe('events', function () {
...
+  })
+
+  it('catch error on connect', function (done) {
+    var pool = new Pool()
+    var connectingErr = 'connecting error'
+    pool.on('connect', function (client) {
+      return client.query('SELECT now()')
+      .then(() => {
+        throw new Error(connectingErr)
+      })
```
Comment (aviaoh) [inline test/events.js]: same for this entire section... it's not connection error but listener/setup error, and `.query` is redundant.

**Review by aviaoh — APPROVED** [review]
Looks great:)
tests need some minor changes + travis build fails (might need to update .travis.yml)

File: test/events.js (hunk 4 above)
Comment (hadasiii) [reply test/events.js]: i want an async function to be called

File: test/events.js (hunk 2 above)
Comment (hadasiii) [reply test/events.js]: fixed

File: test/events.js (hunk 3 above)
Comment (hadasiii) [reply test/events.js]: renamed

File: test/events.js (hunk 5 above)
Comment (hadasiii) [reply test/events.js]: renamed

File: index.js (hunk 1 above)
Comment (hadasiii) [reply index.js]: added

File: test/events.js (hunk 4 above)
Comment (aviaoh) [reply test/events.js]: then go to sleep 😴 . calling query will trigger new sequence of events being emitted that might affect results of this test.

Comment (hadasiii) [reply test/events.js]: sleeping now :)

File: test/events.js [hunk 6]
```diff
@@ -22,6 +22,42 @@ describe('events', function () {
...
+  it.only('waits on connect until finished', function (done) {
+    var pool = new Pool()
+    var emittedClient = false
+    pool.on('connect', function (client) {
+      return new Promise(resolve => {
+        setTimeout(resolve, 50)
+      })
+      .then(() => {
+        emittedClient = client
+      })
```
Comment (aviaoh) [inline test/events.js:34]: indentation still bugs me 🤓 
also, looking at current version you can also use
```
return Promise.delay(50)
    .then(() => emittedClient = client)
```

File: test/events.js [hunk 7]
```diff
@@ -22,6 +22,42 @@ describe('events', function () {
     })
   })
 
+  it.only('waits on connect until finished', function (done) {
```
Comment (aviaoh) [inline test/events.js]: forgotten `.only` :)

File: index.js [hunk 8]
```diff
@@ -97,11 +97,20 @@ Pool.prototype._create = function (cb) {
...
+
+    this.log('client connected')
+    this.emit('connect', client)
+      .then(() => {
+        this.log('came back from connect')
+        cb(null, client)
+      })
+      .catch(err => {
+        this.log('connect listener error:', err)
+        this.pool.destroy(client)
```
Comment (aviaoh) [inline index.js:111]: 👌

**Review by aviaoh — APPROVED** [review]
👍

File: .travis.yml [hunk 9]
```diff
@@ -2,9 +2,6 @@ language: node_js
 
 matrix:
   include:
-    - node_js: "4"
-      addons:
-        postgresql: "9.1"
```
Comment (aviaoh) [inline .travis.yml]: please be aware that this change will prevent original repo to accept this change.
worth add comment here (in this pr) explaining why you removed support of node 4

Comment (hadasiii) [reply .travis.yml]: sure thing

File: test/events.js (hunk 6 above)
Comment (hadasiii) [reply test/events.js:34]: Dont want to use bluebird promises, im gonna keep it simple as is..
